moped-editor/src/views/projects/projectView/ProjectComponents/ComponentForm.js

```diff
@@ -92,7 +92,7 @@ const ComponentForm = ({
 
   const componentOptions = useComponentOptions(optionsData);
   const phaseOptions = usePhaseOptions(optionsData);
-  const { component, phase } = watch(["component", "phase"]);
+  const [component, phase] = watch(["component", "phase"]);
```

Comment:
this was causing `component` to always be `undefined`, and so `isSignalComponent` always `false`.

Reply:
I thought that this returning an array was missing from the docs, but the return is [clearly shown there](https://react-hook-form.com/docs/useform/watch). TIL!

Reply:
👍

also the fact that using object destructuring on an array doesn't throw an error seems like a quintessential javascript bug 🤯